### PR TITLE
Add RequestVcErrorDetail enum

### PR DESCRIPTION
**Pull request by arieljbirnbaum**
### Context

Partially implements [P-609](https://linear.app/litentry/issue/P-609/refactoring-struct-requestvcresultorerror)
Further refactoring of the API is needed, but deferred for now for timing constraints.

### TODO

- [x] Pass CI

### CODE REVIEW

File: tee-worker/app-libs/stf/src/trusted_call_result.rs [hunk 1]
```diff
@@ -102,10 +103,33 @@ pub struct RequestVCResult {
 	pub pre_id_graph_hash: H256,
 }
 
+#[derive(Debug, Encode, Decode, Clone)]
```
Comment (arieljbirnbaum) [inline tee-worker/app-libs/stf/src/trusted_call_result.rs:106]: I _tried_ to incorporate the causing errors in the variants below, but ran into two issues:

1. the respective (mostly 3rd party) error types don't implement the needed traits for `derive` to work here, and
2. clippy complained that the type was "too big" for an `Err` variant, hence the `Box` in the `AssertionBuildFailed` variant :sadpanda:

**Review by BillyWooo — COMMENTED** [review]
LGTM
is there any alignment with client-sdk?

**Comment by arieljbirnbaum** [comment]
> LGTM is there any alignment with client-sdk?

We discussed it earlier with @jonalvarezz (although this PR goes for a smaller change in the end)

File: tee-worker/app-libs/stf/src/trusted_call_result.rs [hunk 2]
```diff
@@ -102,10 +103,34 @@ pub struct RequestVCResult {
...
+	CallSendingFailed(String),
+	ExtrinsicConstructionFailed(String), // Stringified itp_extrinsics_factory::Error
+	ExtrinsicSendingFailed(String),      // Stringified sgx_status_t
+}
+
 #[derive(Debug, Encode, Decode, Clone)]
 pub struct RequestVcResultOrError {
-	pub payload: Vec<u8>,
-	pub is_error: bool,
+	pub result: Result<Vec<u8>, RequestVcErrorDetail>,
```
Comment (jonalvarezz) [inline tee-worker/app-libs/stf/src/trusted_call_result.rs:133]: nice! props for using the `Result` type

**Review by jonalvarezz — COMMENTED** [review]
looking good to me! 

are ts-tests updates coming in a follow-up pull requests? it needs update here, for instance

https://github.com/litentry/litentry-parachain/blob/a4f73b57b594ab070ebe8585b20d1f91630d334e/tee-worker/ts-tests/integration-tests/dr_vc.test.ts#L171

**Comment by arieljbirnbaum** [comment]
> ... oh no -\_- CI is green, so that means that line isn't even exercised :sweat_smile: 

😄 i think it is – the value of the non-existent field is falsy, the condition doesn't meet

**Comment by arieljbirnbaum** [comment]
@jonalvarezz 

> 😄 i think it is – the value of the non-existent field is falsy, the condition doesn't meet

:skull:

File: tee-worker/ts-tests/integration-tests/tsconfig.json [hunk 3]
```diff
@@ -14,6 +14,7 @@
     "ts-node": {
         "esm": true,
         "experimentalResolver": true,
-        "experimentalSpecifierResolution": "node"
+        "experimentalSpecifierResolution": "node",
+        "transpileOnly": true
```
Comment (arieljbirnbaum) [inline tee-worker/ts-tests/integration-tests/tsconfig.json:18]: Adding this to mitigate the `error:NULL` messages from mocha; see [this issue](https://github.com/mochajs/mocha/issues/5085) /cc @jonalvarezz @0xverin

**Review by jonalvarezz — APPROVED** [review]
tested locally via https://github.com/litentry/client-sdk/pull/35's update. 

looking good. thanks, Ariel!

**Pull request merged**